graphql-server/src/index.ts

```diff
@@ -93,6 +91,37 @@ async function run() {
     }),
   );
 
+  app.post(
+    '/remote-schema',
+    (req, res) => {
+      const { key } = req.query;
+      if (!key || key !== process.env.API_KEY) {
+        return res.status(403).send('Access denied');
+      }
+
+      try {
+        const { name, url } = req.body;
+
+        if (!name.match(/^[a-z-]{1,50}$/)) {
+          return res.status(400).send('Invalid name');
+        }
+
+        try {
+          if ((new URL(url)).host !== 'squid.subsquid.io') {
```

Comment:
This will reject the schemas from The Graph. We are still using them for Phala Pill & Phala Pet NFTs. I will replace them soon, but they are currently required in production.

Reply:
My plan is to set them in the file at deployment using the example file. It should just mean we can not add new graph routes using the POST endpoint.

Reply:
Great idea